core/modules/user/user.admin.inc

```diff
@@ -527,7 +527,7 @@ function user_admin_roles($form, $form_state) {
   // Add the "disabled" option to the top of the roles list (makes sure that
   // this is the selected option if the role that was previously set as admin is
   // deleted).
-  array_unshift($roles, t('disabled'));
+  $roles = array(0 => t('disabled')) + $roles;
```

Comment:
why not `$roles = array('disabled' => t('disabled')) + $roles;`?

Reply:
I agree with @jromine about opening this as a separate issue. Backdrop often does this (for example in filter.admin.inc):
`  elseif ($admin_role = config_get('system.core', 'user_admin_role')) {`
meaning that you need a 0 value to disable the admin role.